source/mainwindow.cpp

```diff
@@ -236,7 +241,34 @@ void MainWindow::saveAs(QString x) {
   if (currentFile != "" && !image.isNull())
     image.save(currentFile + "." + x, x.toUpper().toLocal8Bit().constData());
 }
-
+void MainWindow::rename(){
+  if(currentFile != "" && !image.isNull()){
+      bool ok;
+      QString text = QInputDialog::getText(this, tr("Rename"),
+                                           tr("Name for the file:"), QLineEdit::Normal,
+                                            QDir::home().dirName(), &ok);
+
+      //Find position of the last '/' to find directory
+      QString dir = currentFile;
+      QString format;
+      std::reverse(dir.begin(), dir.end());
+      int bslash_position = dir.indexOf("/");
+      int dot_position = dir.indexOf(".");
+      bslash_position = currentFile.size() - bslash_position;
+      for(int i = 0; i < dot_position; i++){
+          format.append(dir[i]);
+      }
+      std::reverse(format.begin(), format.end());
+      dir.clear();
+      for(int i = 0; i < bslash_position; i++){
+          dir.append(currentFile[i]);
+      }
```

Comment:
This block is trying to find the absolute path of the directory of the image right?
You can directly calculate the path of the renamed file as - 
```
QFileInfo file(currentFile);
QString renamed_file = file.absoluteDir().absoluteFilePath("newname.png"); 
qInfo() << renamed_file;
```
So you may directly call image.save(renamed_file).

Reply:
My knowledge about Qt library is very low. Sorry for the inconvenience but if you assign me. I would fix the code immediately.

Reply:
That's totally fine! :)